crates/evm/evm/src/executors/invariant/mod.rs

```diff
@@ -332,7 +332,22 @@ impl<'a> InvariantExecutor<'a> {
         let (invariant_test, invariant_strategy) =
             self.prepare_test(&invariant_contract, fuzz_fixtures)?;
 
+        // Start a timer if timeout is set.
+        let start_time = self.config.timeout.map(|timeout| {
+            (std::time::Instant::now(), std::time::Duration::from_secs(timeout.into()))
+        });
+
         let _ = self.runner.run(&invariant_strategy, |first_input| {
+            // Check if the timeout has been reached.
+            if let Some((start_time, timeout)) = start_time {
```

Comment:
Just realized this should be moved below, where we loop through configured run depth
https://github.com/foundry-rs/foundry/blob/31dd1f77fd9156d09836486d97963cec7f555343/crates/evm/evm/src/executors/invariant/mod.rs#L349-L353
 
otherwise an invariant test with big depth won't stop when timeout but will continue to run until the end

Reply:
Fixed!